### PR TITLE
dumbpad: Add "reverse" layout

### PR DESCRIPTION
**Pull request by cyril279**
* Allows more intuitive manipulation of keymap when "reversible" pcb is used
  in reverse/flipped orientation (controller on the right edge of the board)

## Description

This commit creates an additional layout/keymap pair, to provide a template for when the "reversible" version of the dumbpad is used/populated in a reversed/flipped orientation.  The default keymap can be used, but having a keymap that properly matches the pcb layout is more ideal than trying to mentally keep track of how a partially mirrored layout will map to the pcb.

The additional layout is named "reverse" to complement the name in the [pcb repository](https://github.com/imchipwood/dumbpad/tree/master/reversible)
The original "Layout" is renamed to Layout_default

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dumbpad/keymaps/reverse/config.h [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#pragma once
```
Comment (drashna) [inline keyboards/dumbpad/keymaps/reverse/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dumbpad/keymaps/reverse/config.h (hunk 1 above)
Comment (cyril279) [reply keyboards/dumbpad/keymaps/reverse/config.h:17]: Header added via rebase

**Comment by drashna** [comment]
Since this targets user keymaps, this needs to target the `develop` branch.  You can edit the title to change the target

**Comment by imchipwood** [comment]
This PR is in conflict/duplictes some work done in #9259 (adds support for all dumbpad layouts).

**Comment by cyril279** [comment]
True indeed, I will close this PR, and allow it to be handled by the referenced PR.

**Pull request closed without merging**